core/src/main/java/org/infinispan/persistence/spi/SegmentedAdvancedLoadWriteStore.java

```diff
@@ -158,8 +231,10 @@ default boolean delete(int segment, Object key) {
     * @param fetchMetadata whether the metadata should be included in the marshalled entry
     * @return a publisher that will provide the entries from the store that map to the given segments
     */
-   Publisher<MarshalledEntry<K, V>> publishEntries(IntSet segments, Predicate<? super K> filter, boolean fetchValue,
-         boolean fetchMetadata);
+   default Publisher<MarshallableEntry<K, V>> entryPublisher(IntSet segments, Predicate<? super K> filter, boolean fetchValue,
```

Comment:
I don't think it is, because the 9.4.x store will not compile due to the required import changes.

Reply:
The source of the 9.4.x store wouldn't have to change, it could still return a `Publisher<MarshalledEntry<K, V>>`. But it would have to be recompiled in order to generate the bridge method (even though the erased signature is the same, I think), and we should have binary compatibility as much as possible.